org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/linkedediting/LSPLinkedEditingReconcilingStrategy.java

```diff
@@ -248,6 +252,21 @@ String getValueInRange(LinkedEditingRanges ranges, VerifyEvent event, int offset
 		return null;
 	}
 
+	private boolean rangesContainOffset(LinkedEditingRanges ranges, int offset) {
+		try {
+			for (Range range : ranges.getRanges()) {
+				int start = LSPEclipseUtils.toOffset(range.getStart(), fDocument);
+				int end = LSPEclipseUtils.toOffset(range.getEnd(), fDocument);
+				if (start <= offset && offset <= end) {
+					return true;
+				}
+			}
+		} catch (BadLocationException e) {
+			LanguageServerPlugin.logError(e);
+		}
+		return false;
+	}
+
 	private LinkedPositionGroup toJFaceGroup(LinkedEditingRanges ranges) throws BadLocationException {
```

Comment:
also, in this method I believe we can save the offset computation for the end, and instead of
```
		for (Range range : ranges.getRanges()) {
			int startOffset = LSPEclipseUtils.toOffset(range.getStart(), fDocument);
			int length = LSPEclipseUtils.toOffset(range.getEnd(), fDocument) - startOffset;
			res.addPosition(new LinkedPosition(fDocument, startOffset, length));
		}
		return res;
```
just do
```
		for (Range range : ranges.getRanges()) {
			int startOffset = LSPEclipseUtils.toOffset(range.getStart(), fDocument);
			int length = range.getEnd() - range.getStart();
			res.addPosition(new LinkedPosition(fDocument, startOffset, length));
		}
		return res;
```

Reply:
Yes, you are right, we cannot do the `-` on `Position` elements